src/main/java/org/opensearch/neuralsearch/query/NeuralQueryBuilder.java

```diff
@@ -156,6 +161,7 @@ protected void doWriteTo(StreamOutput out) throws IOException {
         if (isClusterOnOrAfterMinReqVersion(METHOD_PARAMS_FIELD.getPreferredName())) {
             MethodParametersParser.streamOutput(out, methodParameters, MinClusterVersionUtil::isClusterOnOrAfterMinReqVersion);
         }
+        RescoreParser.streamOutput(out, rescoreContext);
```

Comment:
same as above

Reply:
The check is there in [RescoreParser](https://github.com/opensearch-project/k-NN/blob/main/src/main/java/org/opensearch/knn/index/query/parser/RescoreParser.java#L105)